### ISSUE TITLE
There is a problem with the ZoneHVAC:FourPipeFanCoil model not responding to non-zero values for Minimum Cold Water Flow Rate.  The flow can still go to zero and the minimum is not respected during operation. (CR #7853)

### ISSUE THREAD

**Issue by axelstudios** [NREL/EnergyPlus#2633]
## 
#### Description

MJW 31 Aug 2009
User entered non-zero value for Minimum Cold Water Flow Rate in ZoneHVAC:FourPipeFanCoil, but the flow rate goes below this value.  Tried doubling the min flow rate, and setting it to zero, and got exactly the same results.  So, it appears this field has no effect.  Did not test Minimum Hot Water Flow Rate.

Inputs:  7853-*
Weather: Tokyo

[FB 11December2009]
Apparently Plant resets the MassFlowRateMinAvail to zero, thus the minimum flow rates get ignored. Most
likely every component that uses ControlCompOutput will have the same issue. I have changed

```
  ZoneController%MaxSetPoint = Node(ActuatedNode)%MassFlowRateMaxAvail
  ZoneController%MinSetPoint = Node(ActuatedNode)%MassFlowRateMinAvail
```

in GeneralRoutines::ControlCompOutput to

```
  ZoneController%MaxSetPoint =  
      MIN(Node(ActuatedNode)%MassFlowRateMaxAvail,Node(ActuatedNode)%MassFlowRateMax)
  ZoneController%MinSetPoint =
      MAX(Node(ActuatedNode)%MassFlowRateMinAvail,Node(ActuatedNode)%MassFlowRateMin)
```

This does fix the problem. I have run BenchmarkLargeHotelNew_USA_IL_CHICAGO-OHARE.idf for
a full year to test whether intoducing minimum ChW and HW flow rates in the fan coil units causes
any problem. The answer is no - the file runs pretty much the same the minimums zero or >0. The
run does slow down and there are more condenser inlet temperature warnings, but therer are
no new warnings, and the answers are very close.

Note that the code in FanCoilUnits (Sim4PipeFanCoil):

  MaxWaterFlow = FanCoil(FanCoilNum)%MaxColdWaterFlow
  MinWaterFlow = FanCoil(FanCoilNum)%MinColdWaterFlow
  !On the first HVAC iteration the system values are given to the controller, but after that
  ! the demand limits are in place and there needs to be feedback to the Zone Equipment
  If(.not. FirstHVACIteration) Then
     MaxWaterFlow = Node(ControlNode)%MassFlowRateMaxAvail
     MinWaterFlow = Node(ControlNode)%MassFlowRateMinAvail
  End If

is not needed, since the function is performed by CompControlOutput. All other components using
CompControlOutput include the same unneeded code.

MJW 17 Feb 2010
This has been reopened, because we discovered that this is the change tha broke WindACAuto and other runs which are now causing fits in diffs for the DSU work.

[FB 17Feb2010]
The current new code in ControlCompOutput is:

 IF ( (SimCompNum ==3) .AND. ( .NOT. PRESENT(AirMassFlow)) ) THEN
   ZoneController%MaxSetPoint = Node(ActuatedNode)%MassFlowRateMaxAvail
   ZoneController%MinSetPoint = Node(ActuatedNode)%MassFlowRateMinAvail
 ELSE
   ZoneController%MaxSetPoint = MIN(Node(ActuatedNode)%MassFlowRateMaxAvail,Node(ActuatedNode)%MassFlowRateMax)
   ZoneController%MinSetPoint = MAX(Node(ActuatedNode)%MassFlowRateMinAvail,Node(ActuatedNode)%MassFlowRateMin)
 END IF

Note that the SimCompNum check applies to reverse action dampers in terminal units. So the fix only applies to
water coils, hw baseboards and the like.

We believe this code to be correct but it is causing poor control in the heating mode of some files. The code will
be taken out and added to the code being developed for phase 2 of the demand side update. The hope is
that this "better" code will work well with the new plant changes.

[13April2012] FB
Hmm; I forgot to update this CR. I did take another look at this a month or so ago. The reason the min
flow is not taking effect is that the water inlet node flow min is set to zero in HVACWaterCoilComponent.f90.

Call InitComponentNodes(0.d0,WaterCoil(CoilNum)%MaxWaterMassFlowRate, &
                                 WaterCoil(CoilNum)%WaterInletNodeNum,    &
                                 WaterCoil(CoilNum)%WaterOutletNodeNum,   &
                                 WaterCoil(CoilNum)%WaterLoopNum,         &
                                 WaterCoil(CoilNum)%WaterLoopSide,        &
                                 WaterCoil(CoilNum)%WaterLoopBranchNum,   &
                                 WaterCoil(CoilNum)%WaterLoopCompNum 

Note the initial zero argument. To fix this the fan coil would need to pass down a min value to the water coils.
Simple enough; but when I kludged a test of fixing a (pretty high) min for the simulation and used 
FanCoilAutoSize as a test file, the simulation went down to 1 minute time steps and switching the
coil on/off. To fix this we would need to put the fan coil into fan cycling mode when the flow is at
the min. Pretty straightforward, but not a killer high priority.

So in summary, we've figured out what is going on and know how to fix it, but we are too lazy to do anything
about it right now

[FB 17April2012]
And what about other components, such as AirTerminal:SingleDuct:VAV:Reheat? Same issue going on here?
With a pretty high min, would also get the same bouncing performance, I would think.

[BG 3-23-2013]  empirical evidence suggest this is beyond the scope of a CR.  moving to enhancement list, Item is "HVAC_General_2013_02.docx"  check in to starteam EL today
## 

External Ref: Ticket 1387
Addressed in `13.03.31 V8.0.0.006` by Brent Griffith
Last build tested: `09.12.19 V5.0.0.010`
